### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [2]

### PR DESCRIPTION
**Pull request by laadvo**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper functions.

### Description
Added documentation to following files:

- [x]  course/CourseFactory.java
- [x]  course/CourseUtilService.java

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Test Coverage

unchanged

### CODE REVIEW

**Review by dearjasmina — APPROVED** [review]
code lgtm

File: src/test/java/de/tum/in/www1/artemis/course/CourseUtilService.java [hunk 1]
```diff
@@ -603,6 +681,12 @@ public Course addCourseWithModelingAndTextExercise() {
         return course;
     }
 
+    /**
+     * Creates and saves a course with one modeling, one text, and one file upload exercise.
+     * 6
```
Comment (DominikRemo) [inline src/test/java/de/tum/in/www1/artemis/course/CourseUtilService.java:686]: Typo?
```suggestion
```

**Review by DominikRemo — COMMENTED** [review]
LGTM, left one comment

**Review by DominikRemo — APPROVED** [review]
LGTM

**Pull request merged**